scheduler.go

```diff
@@ -171,8 +171,18 @@ func (s *Scheduler) WaitForDependency(txn TxnIndex, blocking_txn TxnIndex) *Cond
 }
 
 func (s *Scheduler) ResumeDependencies(txns []TxnIndex) {
+	if len(txns) == 0 {
+		return
+	}
+	minDependencyIdx := txns[0]
 	for _, txn := range txns {
 		s.txn_status[txn].Resume()
+		if txn < minDependencyIdx {
+			minDependencyIdx = txn
+		}
+	}
+	if minDependencyIdx != InvalidTxnVersion.Index {
+		s.DecreaseExecutionIdx(minDependencyIdx)
```

Comment:
I don't think we need this? 🤔

Reply:
do you mean no need DecreaseExecutionIdx at all

Reply:
can you describe the rational of this change?